### PR TITLE
Add new @versions endpoint

### PR DESCRIPTION
**Pull request by njohner**
We add a new `@versions` endpoint to retrieve the list of versions of a document. This endpoint is basically a copy of the `@history` endpoint with a few modifications:
- It only returns versioning entries (whereas the `@history` endpoint also returns workflow modification entries).
- It is batched
- It always returns the initial version, even when it has not been created now. Because we delay the creation of the initial version in Gever to avoid wasting storage space, we make sure to return the correct information for the initial version in the @versions endpoint even when it wasn't created yet.
- Use the creator as the actor of the initial version in `@versions.` This is necessary because certain initial versions in Gever have the wrong actor. This was already [corrected in the `@history` endpoint](https://github.com/4teamwork/opengever.core/pull/6935)
- It is lazy (uses the shadow history, and does not materialise the whole history as does the `@history` endpoint)
- It only returns relevant data

Note that the `@versions` endpoint is only defined for documents, not for mails. As a side note the `@history` endpoint is broken for mails. But neither endpoint is necessary for mails, so IMO we do not need the `versions` endpoint for mails and we do not need to fix the `@history` endpoint

For https://4teamwork.atlassian.net/browse/CA-1912

Definition of Done: https://4teamwork.atlassian.net/wiki/spaces/CHX4TW/pages/917562/

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- API change:
  - [x] Documentation is updated
  - [x] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- New functionality:
  - [ ] for `document` also works for `mail` -> not defined for mails

### CODE REVIEW

**Review by lukasgraf — CHANGES_REQUESTED** [review]
I have some doubts that this is actually lazy. Batched, yes, but not lazy.

In the classic UI we introduced our own [`LazyHistoryMetadataProxy`](https://github.com/4teamwork/opengever.core/blob/master/opengever/document/browser/versions_tab.py#L177) (years ago) to make sure the table in the versions tab is actually lazy. Before, it was batched, but because of some stupid stuff the content history viewlet does, the entire history still was getting consumed even though we only displayed the first 25 results.

I believe the same thing might be happening here - `content_history_viewlet. fullHistory()` will call [`.revisionHistory()`](https://github.com/plone/plone.app.layout/blob/2.3.20/plone/app/layout/viewlets/content.py#L263-L319). And that will eventually fetch every version, via a [mapped `retrieve`](https://github.com/plone/plone.app.layout/blob/2.3.20/plone/app/layout/viewlets/content.py#L311-L317), even if it's not needed later.

If we're implementing our own `@versions` endpoint, couldn't we just reuse our `LazyHistoryMetadataProxy`?

File: opengever/api/history.py [hunk 1]
```diff
@@ -62,3 +69,106 @@ def reply(self):
...
+            return data
+
+        # Listing historical data (versions only)
+        content_history_viewlet = ContentHistoryViewlet(
+            self.context, self.request, None, None
+        )
+        site_url = getSite().absolute_url()
+        content_history_viewlet.navigation_root_url = site_url
+        content_history_viewlet.site_url = site_url
+        history = content_history_viewlet.revisionHistory()
```
Comment (deiferni) [inline opengever/api/history.py:92]: you will benefit from https://github.com/4teamwork/opengever.core/pull/6981, too. But maybe also add a test to make sure that `@versions` also works when users only have `Reader`.

**Comment by njohner** [comment]
Ok I've completely reworked that PR. We now use the lazy history, as suggested by @lukasgraf . @deif, to your comment, I've added a test, but note that the versions endpoint does not return an empty list for a reader, but instead it shows the version. That is also what we do in the classic UI, and I don't really see a problem with that?

**Review by deiferni — APPROVED** [review]
Please add an entry to changes. The api-changelog is updated, but i could not see the regular changelog entry.

LGTM 👍  otherwise!

File: docs/public/dev-manual/api/versions.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+        "@id": "http://example.org/ordnungssystem/dossier-23/document-123/@versions",
+        "items": [
+            {
+                "@id": "http://example.org/ordnungssystem/dossier-23/document-123/@history/2",
```
[the remark this answers is not in the document]
Comment (njohner) [reply docs/public/dev-manual/api/versions.rst:29]: Yeah you're right. Both return the same thing though :-D. But I changed it

**Pull request merged**